node/messages/src/consensus_id.rs

```diff
@@ -0,0 +1,40 @@
+// Copyright (C) 2019-2023 Aleo Systems Inc.
+// This file is part of the snarkOS library.
+
+// The snarkOS library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkOS library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkOS library. If not, see <https://www.gnu.org/licenses/>.
+
+use super::*;
+use fastcrypto::bls12381::min_sig::{BLS12381PublicKey, BLS12381Signature};
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct ConsensusId {
+    pub public_key: BLS12381PublicKey,
+    pub signature: BLS12381Signature,
+}
+
+impl MessageTrait for Box<ConsensusId> {
+    fn name(&self) -> String {
+        "ConsensusId".to_string()
+    }
+
+    fn serialize<W: Write>(&self, writer: &mut W) -> Result<()> {
+        Ok(bincode::serialize_into(writer, &(self.public_key.clone(), self.signature.clone()))?)
```

Comment:
not too fond of these clones, do we need them for serialization :thinking:?

Reply:
Yeah, unfortunately neither of these types are `Copy`.

Reply:
yeah, but `bincode::serialize_into` expects references